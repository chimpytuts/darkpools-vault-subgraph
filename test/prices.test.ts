import { runTestCases } from './test-runner';
import { gql } from '@apollo/client/core';

const testCases = [
  {
    id: 'getPrices',
    query: gql`
      query {
        tokenPrices {
          id
          asset
          price
          pricingAsset
          amount
          block
        }
      }
    `,
  },
  {
    id: 'getTotalLiquidity',
    query: gql`
      query {
        balancers {
          id
          totalLiquidity
        }
      }
    `,
  },
];

runTestCases('TokenPrices', testCases);
