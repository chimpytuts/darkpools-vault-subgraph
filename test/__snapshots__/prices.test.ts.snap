// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenPrices resolvers getPrices 1`] = `
Object {
  "data": Object {
    "tokenPrices": Array [
      Object {
        "__typename": "TokenPrice",
        "amount": "59361.820196",
        "asset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
        "block": "66",
        "id": "0x52c1245a6e1948424a588a621605a9b88dd50e6f000200000000000000000003-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2-66",
        "price": "593.61820196",
        "pricingAsset": "0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "100",
        "asset": "0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2",
        "block": "66",
        "id": "0x52c1245a6e1948424a588a621605a9b88dd50e6f000200000000000000000003-0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-66",
        "price": "0.001684584463040746480549512966622241",
        "pricingAsset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "3.09945704",
        "asset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
        "block": "63",
        "id": "0x75fe54ce4f6503c8d714d9497639608f591ac479000200000000000000000000-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-0xcd80986f08d776ce41698c47f705cdc99ddbfb0a-63",
        "price": "0.0309945704",
        "pricingAsset": "0xcd80986f08d776ce41698c47f705cdc99ddbfb0a",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "100",
        "asset": "0xcd80986f08d776ce41698c47f705cdc99ddbfb0a",
        "block": "63",
        "id": "0x75fe54ce4f6503c8d714d9497639608f591ac479000200000000000000000000-0xcd80986f08d776ce41698c47f705cdc99ddbfb0a-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-63",
        "price": "32.26371545385252379558711354166729",
        "pricingAsset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "3714.408996305164339837",
        "asset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
        "block": "64",
        "id": "0x80cdbbe17a65107c8f14f38389dfb2fe036907e8000200000000000000000001-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-0xf702269193081364e355f862f2cfbfcdc5db738c-64",
        "price": "37.14408996305164339837",
        "pricingAsset": "0xf702269193081364e355f862f2cfbfcdc5db738c",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "100",
        "asset": "0xf702269193081364e355f862f2cfbfcdc5db738c",
        "block": "64",
        "id": "0x80cdbbe17a65107c8f14f38389dfb2fe036907e8000200000000000000000001-0xf702269193081364e355f862f2cfbfcdc5db738c-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-64",
        "price": "0.02692218334046494154407769927365821",
        "pricingAsset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "59646.403285972368716863",
        "asset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
        "block": "67",
        "id": "0x81b802515b0e8a583fc934bdac4cff214baa09ea000200000000000000000004-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-0x5c0e66606eabec1df45e2add26c5df8c0895a397-67",
        "price": "596.46403285972368716863",
        "pricingAsset": "0x5c0e66606eabec1df45e2add26c5df8c0895a397",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "100",
        "asset": "0x5c0e66606eabec1df45e2add26c5df8c0895a397",
        "block": "67",
        "id": "0x81b802515b0e8a583fc934bdac4cff214baa09ea000200000000000000000004-0x5c0e66606eabec1df45e2add26c5df8c0895a397-0x4cddb3505cf09ee0fa0877061eb654839959b9cd-67",
        "price": "0.001676547025317752621305123801628179",
        "pricingAsset": "0x4cddb3505cf09ee0fa0877061eb654839959b9cd",
      },
      Object {
        "__typename": "TokenPrice",
        "amount": "100",
        "asset": "0xbf92d6171938ebf1c6c14bf436471694f7d282de",
        "block": "65",
        "id": "0xa6128d6807bbbf8530e44ee0f458f5f1f5f9d80d000200000000000000000002-0xbf92d6171938ebf1c6c14bf436471694f7d282de-0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2-65",
        "price": "1.002290872585675920870388675727422",
        "pricingAsset": "0x7c0c5ada758cf764ecd6bac05b63b2482f90bbb2",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
}
`;

exports[`TokenPrices resolvers getTotalLiquidity 1`] = `
Object {
  "data": Object {
    "balancers": Array [
      Object {
        "__typename": "Balancer",
        "id": "2",
        "totalLiquidity": "49303061.89357238770835822117385191",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
}
`;
